packages/core/lib/segment_emitter.js

```diff
@@ -1,12 +1,70 @@
 var dgram = require('dgram');
 
+var batcher = require('atomic-batcher');
 var logger = require('./logger');
 
 var DEFAULT_ADDRESS = '127.0.0.1';
 var DEFAULT_PORT = 2000;
 var PROTOCOL_HEADER = '{"format": "json", "version": 1}';
 var PROTOCOL_DELIMITER = '\n';
 
+function batchRun (work, callback) {
+  var client = dgram.createSocket('udp4');
+
+  processWork(client, work, 0, function () {
+    try {
+      client.close();
+    } finally {
+      callback();
+    }
+  });
+}
+
+function processWork (client, work, index, callback) {
+  if (index >= work.length) {
+    callback();
+    return;
+  }
+
+  sendMessage(client, work[index], function () {
+    processWork(client, work, index+1, callback);
+  });
+}
+
+function sendMessage (client, data, batchCallback) {
+  var msg = data.msg;
+  var offset = data.offset;
+  var length = data.length;
+  var port = data.port;
+  var address = data.address;
+  var callback = data.callback;
+
+  client.send(msg, offset, length, port, address, function(err) {
+    try {
+      callback(err);
+    } finally {
+      batchCallback();
+    }
+  });
+}
+
+function BatchingTemporarySocket() {
+  this.sendBatcher = batcher(batchRun);
+}
+
+BatchingTemporarySocket.prototype.send = function (msg, offset, length, port, address, callback) {
+  var work = {
+    msg: msg,
+    offset: offset,
+    length: length,
+    port: port,
+    address: address,
+    callback: callback
+  };
+
+  this.sendBatcher(work);
```

Comment:
should this be renamed batchSend?

Reply:
Yes, that's a much better name. I'll change it.